src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration.Tests/ConfigurationMockTests.cs

```diff
@@ -224,14 +222,14 @@ public async Task GetBatch()
         [Test]
         public async Task ConfiguringTheClient()
         {
-            var options = ConfigurationClient.CreateDefaultPipelineOptions();
+            var options = new ConfigurationPipelineOptions();
             options.ApplicationId = "test_application";
             options.AddService(ArrayPool<byte>.Create(1024 * 1024 * 4, maxArraysPerBucket: 4), typeof(ArrayPool<byte>));
             options.Transport = new GetMockTransport(s_testSetting.Key, default, s_testSetting, HttpStatusCode.RequestTimeout, HttpStatusCode.OK);
             options.RetryPolicy = RetryPolicy.CreateFixed(2, TimeSpan.FromMilliseconds(100), 408 /* RequestTimeout */);
 
             var testPolicy = new TestPolicy();
-            options.AddPerRetryPolicy(testPolicy);
+            options.AppendPolicies.Add(testPolicy);
```

Comment:
AppendPolicies is a verb phrase. We use noun phrases for property names.

I would rather do: options.Pipeline.AddPolicy(policy, before|after)

Reply:
What would be the type of `Pipeline`?

Reply:
What's the type of AppendPolicies?

Reply:
`IList<HttpPipelinePolicy>`

Reply:
what's the advantage of the design in this PR over methods on options (just like pre this PR)?

Reply:
Conceptually it's the same just doesn't try to hide collections that we have internally anyway, allows customers to remove/clear/replace/reorder policies.

Reply:
OK, in this case can we do (at least for now):
```c#
options.PerCallPolicies.Add(...)
options.PerRetryPolicies.Add(...)
```

Reply:
Sounds good.